### PR TITLE
drivers: npmx: shell: add helper functions for rounding setters

### PR DESCRIPTION
**Pull request by jaz1-nordic**
JIRA: NRFX-4629

test:PR-191

### CODE REVIEW

File: drivers/npmx/npmx_shell.c [hunk 1]
```diff
@@ -361,6 +361,28 @@ static int cmd_charger_charging_current_get(const struct shell *shell, size_t ar
 	return 0;
 }
 
+static bool charger_charging_current_set_helper(const struct shell *shell,
```
Comment (magp-nordic) [inline drivers/npmx/npmx_shell.c:364]: What do you think about creating one function setting and printing read value, that would take set/get functions as arguments?

File: drivers/npmx/npmx_shell.c [hunk 2]
```diff
@@ -649,9 +671,35 @@ static int cmd_charger_trickle_set(const struct shell *shell, size_t argc, char
...
 
-static int cmd_die_temp_set(const struct shell *shell, size_t argc, char **argv,
-			    npmx_error_t (*func)(npmx_charger_t const *p_instance,
-						 int16_t temperature))
+static bool die_temp_set_helper(const struct shell *shell, npmx_charger_t *charger_instance,
+				int16_t temperature, int16_t *p_temperature,
+				npmx_error_t (*func_set)(npmx_charger_t const *p_instance,
+							 int16_t temperature),
+				npmx_error_t (*func_get)(npmx_charger_t const *p_instance,
+							 int16_t *temperature))
```
Comment (magp-nordic) [inline drivers/npmx/npmx_shell.c:679]: Those arguments can be put in a struct, beside the shell, to decrease the number of passed arguments.

**Comment by nordic-segl** [comment]
I see duplicated 'Success' responses:
```
uart:~$ npmx charger ntc_resistance cold set 25228
Info: Set value may be different than requested.
Success: 25189 Ohm.
Success: 25189 ohms.
```

File: drivers/npmx/npmx_shell.c [hunk 3]
```diff
@@ -1144,6 +1144,28 @@ static int cmd_buck_vout_select_get(const struct shell *shell, size_t argc, char
...
+	}
+
+	err_code = npmx_charger_charging_current_get(charger_instance, p_charging_current);
+	if (!check_error_code(shell, err_code)) {
+		print_get_error(shell, "charging current");
+		return false;
+	}
+
+	if (current != *p_charging_current) {
+		shell_info(shell, "Info: Set value may be different than requested.");
```
Comment (masz-nordic) [inline drivers/npmx/npmx_shell.c:1164]: How about
```suggestion
		shell_info(shell, "Info: Requested value was %u but reading will return %u due to approximations.", current, *p_charging_current);
```

Comment (masz-nordic) [reply drivers/npmx/npmx_shell.c:1164]: Same in other functions.

Comment (wilu-nordic) [reply drivers/npmx/npmx_shell.c:1164]: Please create a helper function for this printing.

File: drivers/npmx/npmx_shell.c (hunk 1 above)
Comment (magp-nordic) [reply drivers/npmx/npmx_shell.c]: Bump

**Comment by nordic-segl** [comment]
Waiting for results from nightly tests (build #2):
https://jenkins-ncs.nordicsemi.no/job/HCS/job/NPMX/job/npmx_zephyr_develop_samples/view/change-requests/job/PR-191/

**Review by nordic-segl — APPROVED** [review]

File: drivers/npmx/shell/charger.c [hunk 4]
```diff
@@ -75,9 +93,36 @@ static int cmd_charger_charging_current_get(const struct shell *shell, size_t ar
...
+	}
+
+	err_code = func_get(charger_instance, p_temperature);
+	if (!check_error_code(shell, err_code)) {
+		print_get_error(shell, "die temperature threshold");
+		return false;
+	}
+
+	if (temperature != *p_temperature) {
+		shell_info(shell, "Info: Set value may be different than requested.");
```
Comment (nordic-segl) [inline drivers/npmx/shell/charger.c:116]: I made calculations and tests - for all valid values of die_temp (from -416 *C to +395 *C), get returns exactly the same value which was set.

**Comment by nordic-segl** [comment]
All tests are passing, see build #3 here
https://jenkins-ncs.nordicsemi.no/job/HCS/job/NPMX/job/npmx_zephyr_develop_samples/view/change-requests/job/PR-191/

File: drivers/npmx/npmx_shell.c (hunk 1 above)
Comment (jaz1-nordic) [reply drivers/npmx/npmx_shell.c]: As we discussed offline, I prefer to stay with the current solution due to the problem of passing pointers to functions that differ in arguments.

File: drivers/npmx/npmx_shell.c (hunk 2 above)
Comment (jaz1-nordic) [reply drivers/npmx/npmx_shell.c]: As [here](https://github.com/NordicSemiconductor/npmx-zephyr/pull/110#discussion_r1386598731).

**Review by masz-nordic — APPROVED** [review]

File: drivers/npmx/shell/shell_common.c [hunk 5]
```diff
@@ -246,3 +246,21 @@ bool charger_disabled_check(const struct shell *shell, npmx_charger_t *charger_i
 	}
 	return true;
 }
+
+void value_difference_info(const struct shell *shell, shell_arg_type_t arg_type, uint32_t value_set,
+			   uint32_t value_get)
+{
+	if (value_set != value_get) {
+		if (arg_type == SHELL_ARG_TYPE_INT32_VALUE) {
```
Comment (magp-nordic) [inline drivers/npmx/shell/shell_common.c:254]: switch would be better, but it is not required.

**Review by magp-nordic — APPROVED** [review]

**Pull request merged**